internal/storage/gcs/request.go

```diff
@@ -376,3 +393,22 @@ type DeleteObjectRequest struct {
 	// is not equal to this value.
 	MetaGenerationPrecondition *int64
 }
+
+type ListFoldersRequest struct {
+	// Required. Name of the bucket in which to look for folders. The bucket must
+	// be a hierarchical namespace enabled bucket.
+	Parent string
+	// Optional. Maximum number of folders to return in a single response. The
+	// service will use this parameter or 1,000 items, whichever is smaller.
+	PageSize int32
+	// Optional. A previously-returned page token representing part of the larger
+	// set of results to view.
+	PageToken string
+	// Optional. Filter results to folders whose names begin with this prefix.
+	// If set, the value must either be an empty string or end with a '/'.
+	Prefix string
+	// Optional. If set, returns results in a directory-like mode. The results
+	// will only include folders that either exactly match the above prefix, or
+	// are one level below the prefix. The only supported value is '/'.
+	Delimiter string
+}
```

Comment:
we can pass fields directly instead of wrapper and if we have more than 2 fields then an a struct to hold arguments ahsould be created anyways exact same wrapper is not required here imo, Delimiter can be added from inside bucket handle, can we add other fields also from inside bucket handle (parent, page size and page token) ot all of them are passed via kernel only ?

Reply:
I feel we should align more closely with the existing [ListObject struct](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/internal/fs/inode/dir.go#L591) that we're passing from dir.go. We can't add the Token field, and PageSize from bucket_handle as it is defined in dir.go.

Reply:
Prefix, PageSize, and PageToken are required. I kept Delimiter to align with the ListObject struct.

Reply:
As discussed offline. We are keeping this format only.